test/components/TopNav/ClearRecentTicketsTest.js

```diff
@@ -0,0 +1,180 @@
+import { expect } from 'chai'
+import React from 'react'
+import { Route, Switch } from 'react-router-dom'
+import MenuItem from 'material-ui/MenuItem'
+import ActionDeleteForever from 'material-ui/svg-icons/action/delete-forever'
+
+import { GetMockDispatch, GetDispatchRecorder } from 'test/helpers/mockDispatch'
+import createComponent from 'test/helpers/shallowRenderHelper'
+
+import {
+  clearRecentTicketsButtonAction,
+  ClearRecentTicketsButton,
+  mapStateToClearRecentTicketsProps,
+  ConnectedClearRecentTicketsButton,
+  ClearRecentTickets
+} from 'components/TopNav/ClearRecentTickets'
+import { removePreviousTicketsFromRecent } from 'actions/ticketActions'
+
+
+describe('ClearRecentTickets', function () {
+  describe('clearRecentTicketsButtonAction', function () {
+    context('When currentTicketId is undefined', function () {
+      const dispatchRecorder = GetDispatchRecorder()
+      const dispatch = GetMockDispatch(dispatchRecorder)
+      const initializedAction = clearRecentTicketsButtonAction(dispatch)
+
+      it('Should return a function', function () {
+        expect(initializedAction).to.be.a('function')
+      })
+
+      initializedAction()
+
+      it('Should dispatch removePreviousTicketsFromRecent with no current value', function () {
+        expect(dispatchRecorder.action.type).to.equal('REMOVE_PREVIOUS_TICKETS')
+        expect(dispatchRecorder.action.current).to.be.undefined
+      })
+    })
+
+    context('When currentTicketId has value', function () {
+      const dispatchRecorder = GetDispatchRecorder()
```

Comment:
Dispatching more than one action with the same dispatchRecorder causes dispatchRecorder.action to become an array, because that's how we test thunks; it's not designed to be reusable over multiple dispatch statements.

initializedAction is initialized with different arguments in each of these contexts, and that difference is material to the results of the test.

Reply:
After sleeping on it, I figured out the pattern…

`dispatch` and `dispatchRecorder` should be in a beforeEach block (guaranteeing that they are unique for each test).  This will highlight that the initializedAction is the focus of the test.